TriblerGUI/widgets/trustpage/js/drawing.js

```diff
@@ -28,48 +28,102 @@ function drawNodes(svg, data, on_click) {
-        });
+        .attr("cy", config.node.circle.cy);
 
-    // Append a <text> element to it
+    // Text element in the middle of the node with the last x public key characters
     groups
         .append("text")
-        .attr("x", config.node.publicKeyLabel.x)
-        .attr("y", config.node.publicKeyLabel.y)
+        .attr("dominant-baseline", "central")
+        .attr("text-anchor", "middle")
         .style("font-family", config.node.publicKeyLabel.fontFamily)
         .style("font-size", config.node.publicKeyLabel.fontSize)
+        .style("font-weight", config.node.publicKeyLabel.fontWeight)
         .style("fill", config.node.publicKeyLabel.color)
         .text(function (d) {
-            return d.public_key.substr(-5);
+            return d.public_key.substr(-config.node.publicKeyLabel.characters);
+        });
+
+    // Transparent circle for the mouse events
+    groups
+        .append("circle")
+        .style("fill-opacity", "0")
+        .attr("r", config.node.circle.radius)
+        .attr("cx", config.node.circle.cx)
+        .attr("cy", config.node.circle.cy)
+        .style("cursor", config.node.circle.cursor)
+        .on("click", on_click)
+        .on("mouseover", function (d) {
+            mouseInNode(d, hoverInfoLabel);
+        })
+        .on("mouseout", function () {
+            hoverInfoLabel.style("opacity", 0);
         });
 
     // Return the group of <svg.node>
     return groups;
 }
 
 /**
+ * Show the node information when the user hovers over a node
+ * @param node the node to get the information of
+ * @param infoLabel the label to display the information in
+ */
+function mouseInNode(node, infoLabel) {
+    var quantities = {
+        "Public key" : "..." + node.public_key.substr(node.public_key.length - config.node.hoverLabel.publicKeyCharacters),
+        "Page rank score" : node.page_rank.toFixed(config.node.hoverLabel.pageRankDecimals),
+        "Total uploaded" :  node.total_up + " bytes",
+        "Total downloaded" :  node.total_down + " bytes"
+    };
+
+    infoLabel
+        .style("opacity", config.node.hoverLabel.opacityOnNode)
+        .style("display", "block")
+
+        // Update the label with the information corresponding to the node
+        .html("<table>" + getTableContents(quantities) + "</table>")
+
+        // Compensate for user window width resizing
+        .style("left",
+            (node.x +
+            (document.body.clientWidth - document.getElementById("graph").getBoundingClientRect().width) / 2) + "px")
+        .style("top", node.y + "px");
+}
+
+function getTableContents(quantities) {
+    var tableRows = "";
+    for (var quantity in quantities)
+        tableRows.concat(createTableRow(quantity.key, quantity.value));
+    return tableRows;
+}
+
+function createTableRow(quantityDescription, quantityValue) {
+    return "<tr>" +
+        "<td class=\"metric\"> + quantityDescription + </td>" +
+        "<td> + quantityValue + </td>" +
+        "</tr>";
+}
```

Comment:
This is a bit inconsistent with the above. You can use d3 for creating html. Something like this, but then more:
```
    var rows = d3.select("table").selectAll("tr").data(quantities)
    .exit().remove() // this removes all old table rows
    .enter().append('tr') // this makes new table rows for the new data
    rows.append('td') // append a cell
    .attr('class','metric')
    .html(function(d){return d.key})
```